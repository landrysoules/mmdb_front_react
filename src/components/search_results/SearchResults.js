import React from 'react'
import ImdbResultsContainer from './ImdbResultsContainer'

const SearchResults = ({searchResults}) => {
  if (searchResults.searchMovies.error) {
    return (
      <h2>{searchResults.searchMovies.error.data}</h2>
    )
  } else {
    return (
      <div className="row">
      <div className="col-lg-12">
      <div className="row">
      <table>
        <tbody>
          {searchResults.searchMovies.localMovies.map(movie => {
            if (movie.name) {
              return (
                <tr key={movie._id.$oid}>
                  <td>{movie.name}</td>
                  <td>{movie.year}</td>
                </tr>
              )
            }
            return ''
          })}
        </tbody>
      </table>
    </div>
    {/* <ImdbResultsContainer movies={ searchResults.searchMovies.imdbMovies }/> */}
    <ImdbResultsContainer/>
  </div>
  </div>

    )
  }
}
export default SearchResults;
