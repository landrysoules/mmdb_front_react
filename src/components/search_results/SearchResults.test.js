import React from 'react'
import renderer from 'react-test-renderer'
import SearchResults from './SearchResults'
describe('SearchResults', () => {
  it('display error message when API returns an error', () => {
    const component = renderer.create(<SearchResults searchResults={{
      searchMovies: {
      error: {data: "internal server error"}
    }
    }}/>)
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
