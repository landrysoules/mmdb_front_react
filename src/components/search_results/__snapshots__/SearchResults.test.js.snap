// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResults display error message when API returns an error 1`] = `
<h2>
  internal server error
</h2>
`;
